### DISCUSSION TITLE
Carousel of models not working. Problem with switcSrc

### DISCUSSION THREAD

**Discussion by Hexa93** [google/model-viewer#4097] [Q&A] [answered]
### Discussed in https://github.com/google/model-viewer/discussions/4092

<div type='discussions-op-text'>

<sup>Originally posted by **Hexa93** February  3, 2023</sup>
Hello everyone. I'm having a hard time trying to find out why the file directory won't work. I've followed [this example](https://modelviewer.dev/examples/augmentedreality/index.html) with a few tweaks. My problem is with the slider buttons at the bottom, when I try to change the file, the site freeze (i.e. switchSrc doesn't switch the src).
I've used this function:

 window.switchSrc = (element, name) => {
    const base = "C:/Users/User/Desktop/products" + name;
    modelViewer.src = base + '.glb';
    modelViewer.poster = base + '.webp';
    const slides = document.querySelectorAll(".slide");
    slides.forEach((element) => {element.classList.remove("selected");});
    element.classList.add("selected");
  };

As well as this one:

const baseURL = "C:/Users/User/Desktop/products/";

window.switchSrc = (element, name) => {
  const src = baseURL + name + '.glb';
  const poster = baseURL + name + '.webp';
  modelViewer.src = src;
  modelViewer.poster = poster;
  const slides = document.querySelectorAll(".slide");
  slides.forEach((element) => {element.classList.remove("selected");});
  element.classList.add("selected");
};

Neither seems to work. It is not reading the directory correctly.

Thank you in advance for any further help.</div>

**Comment by dzhakhangirov** [comment]
I think it can't work locally. The files must be on the server. https://arguv.com/ar_viewer/custom/0/slider/

**Comment by Hexa93** [comment]
> I think it can't work locally. The files must be on the server. https://arguv.com/ar_viewer/custom/0/slider/

I uploaded it to my server. Still no success.
Here's the link:
https://www.cobystudio.it/Carosello/

Also, for further examination, here's my code:

<!doctype html>
<html lang="en">
  <head>
    <title>&lt;model-viewer&gt; template</title>
    <meta charset="utf-8">
    <meta name="description" content="&lt;model-viewer&gt; template">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link type="text/css" href="./styles.css" rel="stylesheet"/>
    
    <script src="https://unpkg.com/focus-visible@5.0.2/dist/focus-visible.js" defer></script>
  </head>
  <body>
    
    <model-viewer src="models/coffe.glb" poster="poster.webp" shadow-intensity="1" ar camera-controls touch-action="pan-y" alt="A 3D model carousel">
      <div class="progress-bar hide" slot="progress-bar">
          <div class="update-bar"></div>
      </div>
      <button slot="ar-button" id="ar-button">
          View in your space
      </button>
      <div id="ar-prompt">
          <img src="ar_hand_prompt.png">
      </div>
      <button id="ar-failure">
        AR is not tracking!
      </button>
    
      <div class="slider">
        <div class="slides">
          <button class="slide selected" onclick="switchSrc(this, 'Cafe')" style="background-image: url('Cafe.webp');">
    
          </button><button class="slide" onclick="switchSrc(this, 'Visore')" style="background-image: url('Visore.webp');">
        </button></div>
      </div>
    </model-viewer>  
    <script type="module">
      const modelViewer = document.querySelector("model-viewer");
      
      window.switchSrc = (element, name) => {
        const base = "./models" + name;
  modelViewer.src = base + '.glb';
  modelViewer.poster = base + '.webp';
  const slides = document.querySelectorAll(".slide");
  slides.forEach((element) => { element.classList.remove("selected"); });
  element.classList.add("selected");
};
    
      document.querySelector(".slider").addEventListener('beforexrselect', (ev) => {
        // Keep slider interactions from affecting the XR scene.
        ev.preventDefault();
      });
    </script>
    
    <script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>
  </body>
</html>

**Comment by Hexa93** [comment]
I've also changed the absolute path to a relative path, such as "./models", so it should not fetch from local directory. But still no succes.

**Comment by dzhakhangirov** [comment]
![image](https://user-images.githubusercontent.com/124194927/216602629-663096d4-4636-4854-84c4-175081cbf8f8.png)

**Comment by dzhakhangirov** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/124194927/216603596-2c9851d4-5813-4fa7-ad09-1b7cc02eced9.png)

**Comment by dzhakhangirov** [comment]
Google Chrome F12

**Comment by Hexa93** [comment]
Thank you very much, that solved my problem. Even thought it doesn't make sense to me, because thoese names are not in my code, so I don't know where and why it is trying to fetch them. But anyway, thanks again.

**Comment by dzhakhangirov** [comment]
You are welcome. Your models are a bit heavy. Very long loading. It is advisable to fit into 5-10 MB for display on the web
